src/emitter/rewriting/visiting_emitter.rs

```diff
@@ -313,18 +330,59 @@ impl<'a, 'b, 'c, 'd> VisitingEmitter<'a, 'b, 'c, 'd> {
             }
         }
     }
+    fn set_curr_loc(&mut self) {
+        let loc = match self.app_iter.curr_loc() {
+            OrcaLocation::Module {
+                func_idx,
+                instr_idx,
+                ..
+            }
+            | OrcaLocation::Component {
+                func_idx,
+                instr_idx,
+                ..
+            } => (func_idx as i32, instr_idx as i32),
+        };
+        //set the current location in bytecode and load some new globals for potential report vars
+        self.report_var_metadata.set_loc(
+            self.curr_script_id.clone(),
+            loc,
+            self.curr_probe_id.clone(),
+            self.curr_num_reports,
+        );
+    }
 }
-impl Emitter for VisitingEmitter<'_, '_, '_, '_> {
+impl Emitter for VisitingEmitter<'_, '_, '_, '_, '_, '_> {
     fn emit_body(&mut self, body: &mut [Statement]) -> Result<bool, Box<WhammError>> {
         let mut is_success = true;
+        for _ in 0..self.curr_num_reports {
+            let default_global = whamm_type_to_wasm(&DataType::I32);
+            let gid = self.app_iter.module.add_global(default_global);
+            self.report_var_metadata
+                .available_i32_gids
+                .push(gid as usize);
+        }
         for stmt in body.iter_mut() {
             is_success &= self.emit_stmt(stmt)?;
+            //now emit the changes to the report vars if needed
```

Comment:
What does this comment mean? Didn't you already emit the changes to report vars through emitting statements?

Reply:
This called func is truly just printing right?

Reply:
It emits the call to the print function, changed the comment to better reflect that